.github/workflows/publish-to-pypi.yml

```
@@ -0,0 +1,39 @@
+name: Publish Modin wheel to PyPI
+
+on:
+  schedule:
+    - cron: "42 0 * * WED"
+  push:
+  workflow_dispatch:
+
+jobs:
+  build-n-publish:
+    name: Build and publish Modin wheel to PyPI
+    environment: release
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        fetch-tags: true
+    - name: Set up Python
```

We need to check out the latest tag because that might not be where `head` is (for point releases and if new commits are merged after the tag)

```suggestion
    - name: Checkout latest git tag
      - if: github.event_name == "push"
      git checkout $(git describe --tags "$(git rev-list --tags --max-count=1)")
    - name: Set up Python
```